### PR TITLE
Use CoordsXY on ViewportInteraction.cpp

### PR DESCRIPTION
**Pull request by tupaschoal**

### CODE REVIEW

File: src/openrct2-ui/interface/ViewportInteraction.cpp [hunk 1]
```diff
@@ -527,14 +527,14 @@ static void viewport_interaction_remove_footpath(TileElement* tileElement, int32
     if (w != nullptr)
         footpath_provisional_update();
 
-    tileElement2 = map_get_first_element_at(x / 32, y / 32);
+    tileElement2 = map_get_first_element_at(mapCoords.x / 32, mapCoords.y / 32);
```
Comment (duncanspumpkin) [inline src/openrct2-ui/interface/ViewportInteraction.cpp]: is there not an overload that takes `CoordsXY`

File: src/openrct2-ui/interface/ViewportInteraction.cpp [hunk 2]
```diff
@@ -544,39 +544,39 @@ static void viewport_interaction_remove_footpath(TileElement* tileElement, int32
...
+void viewport_interaction_remove_park_entrance(TileElement* tileElement, CoordsXY mapCoords)
 {
     int32_t rotation = tileElement->GetDirectionWithOffset(1);
     switch (tileElement->AsEntrance()->GetSequenceIndex())
     {
         case 1:
-            x += CoordsDirectionDelta[rotation].x;
-            y += CoordsDirectionDelta[rotation].y;
+            mapCoords.x += CoordsDirectionDelta[rotation].x;
+            mapCoords.y += CoordsDirectionDelta[rotation].y;
```
Comment (duncanspumpkin) [inline src/openrct2-ui/interface/ViewportInteraction.cpp]: Use the CoordsXY += overload

File: src/openrct2-ui/interface/ViewportInteraction.cpp [hunk 3]
```diff
@@ -544,39 +544,39 @@ static void viewport_interaction_remove_footpath(TileElement* tileElement, int32
...
             break;
         case 2:
-            x -= CoordsDirectionDelta[rotation].x;
-            y -= CoordsDirectionDelta[rotation].y;
+            mapCoords.x -= CoordsDirectionDelta[rotation].x;
+            mapCoords.y -= CoordsDirectionDelta[rotation].y;
             break;
```
Comment (duncanspumpkin) [inline src/openrct2-ui/interface/ViewportInteraction.cpp]: as above

File: src/openrct2-ui/interface/ViewportInteraction.cpp (hunk 1 above)
Comment (tupaschoal) [reply src/openrct2-ui/interface/ViewportInteraction.cpp]: There is not, just an overload for `map_get_surface_element_at`

**Comment by tupaschoal** [comment]
Dealt with issues

**Comment by tupaschoal** [comment]
Is this ok to be merged?

**Pull request merged**